pkg/archive/archive.go

```diff
@@ -382,6 +382,56 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 	return nil
 }
 
+// Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
```

Comment:
ok. Another option is to move the archive Matches into fileutils.go, but call it "OptimizedMatches" or something, then have Matches() do all of the clean-up stuff before calling OptimizedMatches().  But archive calls OptimizedMatches. Then we don't dup code.  just a thought.